lib/ims/lti/models/messages/message.rb

```diff
@@ -73,20 +85,8 @@ def supers_params(instance_variable)
     add_required_params :lti_message_type, :lti_version
 
     def self.generate(params)
-      case params['lti_message_type']
-        when BasicLTILaunchRequest::MESSAGE_TYPE
-          BasicLTILaunchRequest.new(params)
-        when RegistrationRequest::MESSAGE_TYPE
-          RegistrationRequest.new(params)
-        when ContentItemSelectionRequest::MESSAGE_TYPE
-          ContentItemSelectionRequest.new(params)
-        when ContentItemSelection::MESSAGE_TYPE
-          ContentItemSelection.new(params)
-        when ToolProxyReregistrationRequest::MESSAGE_TYPE
-          ToolProxyReregistrationRequest.new(params)
-        else
-          self.new(params)
-      end
+      klass = self.descendants.select{|d| d::MESSAGE_TYPE == params['lti_message_type']}.first
+      klass ? klass.new(params) : self.new(params)
```

Comment:
I like that idea. It makes the code easier to understand, since at a glance you wouldn't be able to tell what self could be.